README.md

```diff
@@ -1511,3 +1516,200 @@ FunctionDeclaration(path) {
 # Best Practices
 
 > I'll be working on this section over the coming weeks.
+
+## Avoid traversing the AST as much as possible
+
+Traversing the AST is expensive, and it's easy to accidentally traverse the AST
+more than necessary. This could be thousands if not tens of thousands of
+extra operations.
+
+Babel optimizes this as much as possible, merging visitors together if it can in
+order to do everything in a single traversal.
+
+### Merge visitors whenever possible
+
+When writing visitors, is may be tempting to call `path.traverse` in multiple
+places where they are logically necessary.
+
+```js
+path.traverse({
+  Identfier(path) {
+    // ...
+  }
+});
+
+path.traverse({
+  BinaryExpression(path) {
+    // ...
+  }
+});
+```
+
+However, it is far better to write these as a single visitor that only gets run
+once. Otherwise you are traversing the same tree multiple times for no reason.
+
+```js
+path.traverse({
+  Identfier(path) {
+    // ...
+  },
+  BinaryExpression(path) {
+    // ...
+  }
+});
+```
+
+### Do not traverse when manual lookup will do
+
+It may also be tempting to call `path.traverse` when looking for a particular
+node type.
+
+```js
+const visitorOne = {
+  Identfier(path) {
+    // ...
+  }
+};
+
+const MyVisitor = {
+  FunctionDeclaration(path) {
+    path.get('params').traverse(visitorOne);
+  }
+};
+```
+
+However, if you are looking for something specific and shallow. There's a good
+chance you can manually lookup the nodes you need without performing a costly
+traversal.
+
+```js
+const MyVisitor = {
+  FunctionDeclaration(path) {
+    path.node.params.forEach(function() {
+      // ...
+    });
+  }
+};
+```
+
+## Optimizing nested visitors
+
+When you are nesting visitors, it might make sense to write them nested in your
+code.
+
+```js
+const MyVisitor = {
+  FunctionDeclaration(path) {
+    path.traverse({
+      Identfier(path) {
+        // ...
+      }
+    });
+  }
+};
+```
+
+However, this creates a new visitor object everytime `FunctionDeclaration()` is
+called above. Which Babel then needs to explode and validate every single time.
+This can be costly, so it is better to hoist the visitor up.
+
+```js
+const visitorOne = {
+  Identfier(path) {
+    // ...
+  }
+};
+
+const MyVisitor = {
+  FunctionDeclaration(path) {
+    path.traverse(visitorOne);
+  }
+};
+```
+
+If you need some state within the nested visitor, like so:
+
+```js
+const MyVisitor = {
+  FunctionDeclaration(path) {
+    var exampleState = path.node.params[0].name;
+
+    path.traverse({
+      Identfier(path) {
+        if (path.node.name === exampleState) {
+          // ...
+        }
+      }
+    });
+  }
+};
+```
+
+You can pass it in as state to the `traverse()` method and have access to it on
```

Comment:
I *think* `You` should be lowercase since it's a continuation of the sentence above the code example